### ISSUE TITLE
Does the RestMixin handle responses?

### ISSUE THREAD

**Issue by thxmike** [agmen-hu/node-datapumps#12]
I have the following sample code which uses the postJson function. Since this is using Restler, I am wiring up to the events from Restler. However, it appears the events are swallowed, since there appears to be no way to get to the response data. I have also tried this with the post function as well. This is required to get response information for a latter operation.

data_pump
        .mixin(datapumps.mixin.RestMixin)
        .process(function (data) {

            var test_record = <some json>
            var test_record_json = JSON.stringify(test_record);

            this.postJson('http://qa71-app.mia.ucloud.int/ucc/api/candidates', { 
                multipart: false, 
                headers: { 'Content-Type' : 'application/json',  'X-ApiKey' : 'xxxxxx'  }, 
                data: test_record_json
            }).on('complete', function(data, response) {
                console.log(response.statusCode);
            }).on('error', function(err, response){
                console.log(response.statusCode)
            }).on('fail', function(err, response){
                console.log(response.statusCode)
            }).on('success', function(err, response){
                console.log(response.statusCode)
            }).on('2XX', function(err, response){
                console.log(response.statusCode)
            }).on('3XX', function(err, response){
                console.log(response.statusCode)
            }).on('4XX', function(err, response){
                console.log(response.statusCode)
            }).on('5XX', function(err, response){
                console.log(response.statusCode)
            }).on('end', function(err, response){
                console.log(response.statusCode)
            });
        })
        .start();

**Comment by novaki** [comment]
Those methods are wrapped to return a Promise instead of an EventEmitter.

You should try this in process:
```js
return this.postJson(...)
  .then(function(responseData) {
    return this.copy(...);
  }.bind(this));
```

Please note that process should return a Promise, that's why we need to return the result of .postJson() and .copy() too.

**Comment by thxmike** [comment]
OK, I apologize but I am not following. I first tried the following to see if the data I am anticipating is correct:

this.postJson(target_service_end_point, { 
                multipart: false, 
                headers: { 'Content-Type' : 'application/json',  'X-ApiKey' :   'xxxxxx' }, 
                data: test_record_json
            }).then(function(responseData) {

                console.log(responseData);
                console.log(responseData.statusCode);
                this.copy(responseData);

            }.bind(this));

I see the response payload, however, I can see the statusCode or header information.

I then tried based on what you stated about process returning a promise and did the following:
first this
data_pump
        .mixin(datapumps.mixin.RestMixin)
        .process(function (data) {
            var test_record = '<some formatted json>'
            var test_record_json = JSON.stringify(test_record);

            return this.postJson(target_service_end_point, { 
                multipart: false, 
                headers: { 'Content-Type' : 'application/json',  'X-ApiKey': 'XXXXXXXX' }, 
                data: test_record_json
            }).then(function(responseData) {
                //console.log(responseData);
                //console.log(responseData.statusCode);
                return this.copy(responseData);
            }.bind(this));
          
        }).start().whenFinished().then(function(data){
            console.log(data);
            console.log(data.statusCode);
            console.log(data.headers);
        });
I get nothing in the console

then I tried this
data_pump
        .mixin(datapumps.mixin.RestMixin)
        .process(function (data) {
            var test_record = '<some formatted json>'
            var test_record_json = JSON.stringify(test_record);

            return this.postJson(target_service_end_point, { 
                multipart: false, 
                headers: { 'Content-Type' : 'application/json',  'X-ApiKey': 'XXXXXXXX' }, 
                data: test_record_json
            }).then(function(responseData) {
                //console.log(responseData);
                //console.log(responseData.statusCode);
                return this.copy(responseData);
            }.bind(this));
          
        }).then(function(data){
            console.log(data);
            console.log(data.statusCode);
            console.log(data.headers);
        }).start(); 
ERROR

then I tried this
data_pump
        .mixin(datapumps.mixin.RestMixin)
        .process(function (data) {
            var test_record = '<some formatted json>'
            var test_record_json = JSON.stringify(test_record);

            return this.postJson(target_service_end_point, { 
                multipart: false, 
                headers: { 'Content-Type' : 'application/json',  'X-ApiKey': 'XXXXXXXX' }, 
                data: test_record_json
            }).then(function(responseData) {
                //console.log(responseData);
                //console.log(responseData.statusCode);
                return this.copy(responseData);
            }.bind(this));
          
        }).start().then(function(data){
            console.log(data);
            console.log(data.statusCode);
            console.log(data.headers);
        }); 
ERROR

**Comment by thxmike** [comment]
After some more digging, I think I see why it does not return all of the data.
In the RestMixin, line 61
You have
return restler[methodName].apply(restler, methodArgs).on('complete', function(result) {

However in Restler's documentation for events,(complete: function(result, response)) 'complete' has two arguments.
The first is the result, which is what you are returning, the second which is missing is the response with all of the statusCodes and header information.

We need both

**Comment by thxmike** [comment]
Any update on this item?

**Comment by novaki** [comment]
I've fixed it in 0.3.26, you should receive the response in the second argument of promise resolved callback:
```js
return this.postJson(...)
  .then(function(result, response) {
    return this.copy(...);
  }.bind(this));
```
